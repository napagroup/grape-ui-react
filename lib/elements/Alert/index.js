"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _component = require("./component");

_forEachInstanceProperty(_context = _Object$keys(_component)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _component[key];
    }
  });
});

var _styled = require("./styled");

_forEachInstanceProperty(_context2 = _Object$keys(_styled)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _styled[key];
    }
  });
});