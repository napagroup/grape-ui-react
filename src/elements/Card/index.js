export * from './component';
export * from './styled';
