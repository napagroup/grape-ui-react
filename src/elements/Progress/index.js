export * from './getProgress';
export * from './styled';
