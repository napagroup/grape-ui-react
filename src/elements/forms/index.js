export * from './CheckboxField';
export * from './DateField';
export * from './Form';
export * from './SelectField';
export * from './TextField';
