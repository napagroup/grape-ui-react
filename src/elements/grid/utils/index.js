import propTypes from '@styled-system/prop-types';
import { getProgressPropTypes } from 'src/elements/Progress';

export const defaultFlexBoxStylesPropsToTrim = [
  ...Object.keys(propTypes.background),
  ...Object.keys(propTypes.border),
  ...Object.keys(propTypes.flexbox),
  ...Object.keys(propTypes.grid),
  ...Object.keys(propTypes.layout),
  ...Object.keys(propTypes.position),
  ...Object.keys(propTypes.shadow),
  ...Object.keys(propTypes.space),
  ...Object.keys(getProgressPropTypes),
  'boxSizing',
];
