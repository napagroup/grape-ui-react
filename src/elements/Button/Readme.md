`<Button>` is grape-ui's button component. It will translate to either a `<button>`, `<Link>`, or `<a>` element depending on what properties are given to it.

```jsx in Markdown
import { BrowserRouter as Router } from 'react-router-dom';
import { Button } from 'src/elements/Button'; // ... from 'grape-ui-react'
import { Flex } from 'src/elements/grid'; // ... from 'grape-ui-react'
import { Code, Text } from 'src/elements/typography'; // ... from 'grape-ui-react'

<Flex
  alignItems="flex-start"
  flexDirection="column"
>
  <Button>
    {'Basic Button '}
    <Code
      code="<button>"
      language="html"
    />
  </Button>
  <Button href="#">
    {'Basic Button '}
    <Code
      code="<a>"
      language="html"
    />
  </Button>
  <Button
    emailHref={{
      to: 'email@grape-ui.com'
    }}
  >
    {'Basic Button using emailHref '}
    <Code
      code="<a>"
      language="html"
    />
  </Button>
  <Router>
    <Button to="./">
      {'Basic Button '}
      <Code
        code="<Link>"
      />
      *
    </Button>
    <Text as="span" color="gray" sm>
      {'*Please note, '}
      <Code
        code="<Button>"
      />
      {' components that use '}
      <Code
        code="<Link>"
      />
      {' need to be wrapped in a '}
      <Code
        code="<Router>"
      />
      {` component from 'react-router-dom'.`}
    </Text>
  </Router>
</Flex>
```

#### Kitchen Sink Usage

```jsx in Markdown
import { useState } from 'react';
import { Button } from 'src/elements/Button'; // ... from 'grape-ui-react'
import { Flex } from 'src/elements/grid'; // ... from 'grape-ui-react'

const [count, setCount] = useState(0);

<Flex
  alignItems="center"
  flexDirection={['column', 'row']}
>
  <Button
    bg="white"
    bgActiveColor="pink"
    bgHoverColor="pink.light"
    border="3px solid #fdd7e4"
    borderRadius="9px"
    color="white"
    contained
    ellipsis
    fontSize="26px"
    m={[1, 2, 3, 4]}
    name="exampleColor"
    onClick={e => this.console.log('Oink')}
    outline
    raised
    title="Click on me and check out the console."
  >
    üê∑
  </Button>
  <Button
    bg="white"
    bgActiveColor="blue"
    bgHoverColor="blue.light"
    border="3px solid cornflowerblue"
    borderRadius="12px"
    color="cornflowerblue"
    contained
    fontSize="12px"
    leadingIcon="üç™"
    m={[1, 2, 3, 4]}
    name="exampleColor"
    onClick={() => setCount(count + 1)}
    outline
    pb={2}
    pl={3}
    pr={3}
    pt={2}
    raised
    textAlign="center"
    title="Click on me and check out the console."
  >
    This cookie button has been clicked {`${count} time${count == 1 ? '' : 's'}`}
  </Button>
</Flex>
```

### Variants

Instead of typing in the background and foreground colors manually, you can utilize `<Button>`'s `variant` attribute. You can also combine this with `<ThemeProvider>` to fully define direct variant colors.

```jsx in Markdown
import { ThemeProvider } from 'styled-components';
import { Button } from 'src/elements/Button'; // ... from 'grape-ui-react'
import { Flex } from 'src/elements/grid'; // ... from 'grape-ui-react'
import * as styledHelpers from '../../utils/styledHelpers'; // ... import { styledHelpers} from 'grape-ui-react'

const theme = {
  buttons: {
    ...styledHelpers.buttonThemes(),
  },
};

<ThemeProvider theme={theme}>
  <Flex flexWrap="wrap" my={[2, 3]}>
    <Button variant="default">
      Default
    </Button>
    <Button variant="primary">
      Primary
    </Button>
    <Button variant="danger">
      Danger
    </Button>
    <Button variant="dark">
      Dark
    </Button>
    <Button variant="info">
      Info
    </Button>
    <Button variant="light">
      Light
    </Button>
    <Button variant="link">
      Link
    </Button>
    <Button variant="secondary">
      Secondary
    </Button>
    <Button variant="success">
      Success
    </Button>
    <Button variant="warning">
      Warning
    </Button>
  </Flex>
  <Flex flexWrap="wrap" my={[2, 3]}>
    <Button variant="contained-default">
      Contained-Default
    </Button>
    <Button variant="contained-primary">
      Contained-Primary
    </Button>
    <Button variant="contained-danger">
      Contained-Danger
    </Button>
    <Button variant="contained-dark">
      Contained-Dark
    </Button>
    <Button variant="contained-info">
      Contained-Info
    </Button>
    <Button variant="contained-light">
      Contained-Light
    </Button>
    <Button variant="contained-link">
      Contained-Link
    </Button>
    <Button variant="contained-secondary">
      Contained-Secondary
    </Button>
    <Button variant="contained-success">
      Contained-Success
    </Button>
    <Button variant="contained-warning">
      Contained-Warning
    </Button>
  </Flex>
</ThemeProvider>
```

#### Custom Styles

```jsx in Markdown
import { ThemeProvider } from 'styled-components';
import { Button } from 'src/elements/Button'; // ... from 'grape-ui-react'
import { Flex } from 'src/elements/grid'; // ... from 'grape-ui-react'
import * as styledHelpers from '../../utils/styledHelpers'; // ... import { styledHelpers} from 'grape-ui-react'
import { getGlobalOverrides, getGlobalStyles } from 'src/global-styles'; // ... from 'grape-ui-react'

const globalStyles = getGlobalStyles();
const { colors } = globalStyles;

const colorThemeProps = {
  theme: {
    colors: {
      brandDanger: {
        base: 'hsl(3.5, 90.3%, 51.4%)',
        dark: 'hsl(3.5, 90.3%, 41.4%)',
        light: 'hsl(3.5, 90.3%, 61.4%)',
      },
      brandDark: {
        base: 'hsl(110, 5.5%, 21.6%)',
        dark: 'hsl(110, 5.5%, 11.6%)',
        light: 'hsl(110, 5.5%, 31.6%)',
      },
      brandInfo: {
        base: 'hsl(170.6, 16.8%, 62.7%)',
        dark: 'hsl(170.6, 16.8%, 52.7%)',
        light: 'hsl(170.6, 16.8%, 72.7%)',
      },
      brandLight: {
        base: 'hsl(220.6, 100%, 93.3%)',
        dark: 'hsl(220.6, 100%, 83.3%)',
        light: 'hsl(220.6, 100%, 100%)',
      },
      brandLink: {
        base: 'hsl(146.5, 100%, 23.5%)',
        dark: 'hsl(146.5, 100%, 13.5%)',
        light: 'hsl(146.5, 100%, 33.5%)',
      },
      brandLinkHover: {
        base: 'hsl(87.4, 68.3%, 47.1%)',
        dark: 'hsl(87.4, 68.3%, 37.1%)',
        light: 'hsl(87.4, 68.3%, 57.1%)',
      },
      brandPrimary: {
        base: 'hsl(233.31, 85.61%, 29.98%)',
        dark: 'hsl(233.91, 34.91%, 24.71%)',
        light: 'hsl(233.81, 68.81%, 48.51%)',
      },
      brandSecondary: {
        base: 'hsl(53.1, 85.6%, 30%)',
        dark: 'hsl(53.1, 85.6%, 20%)',
        light: 'hsl(53.1, 85.6%, 40%)',
      },
      brandSuccess: {
        base: 'hsl(236, 17.8%, 49.6%)',
        dark: 'hsl(236, 17.8%, 39.6%)',
        light: 'hsl(236, 17.8%, 59.6%)',
      },
      brandWarning: {
        base: 'hsl(53.3, 100%, 47.1%)',
        dark: 'hsl(53.3, 100%, 37.1%)',
        light: 'hsl(53.3, 100%, 57.1%)',
      },
    },
  },
};
const globalOverrides = getGlobalOverrides(colorThemeProps);

const theme = {
  buttons: {
    ...styledHelpers.buttonThemes(null, globalOverrides),
  },
  border: {
    borderRadius: {
      base: '32px',
      lg: '64px',
      sm: '16px',
    },
  },
  colors: {
    ...colors,
    brandPrimary: {
      base: 'hsl(233.31, 85.61%, 29.98%)',
      dark: 'hsl(233.91, 34.91%, 24.71%)',
      light: 'hsl(233.81, 68.81%, 48.51%)',
    },
    yellow: {
      base: 'hsl(65, 65%, 50%)',
      dark: 'hsl(65, 65%, 25%)',
      light: 'hsl(65, 65%, 68%)',
    },
  },
  fonts: {
    base: 'Tahoma, Verdana, Segoe, sans-serif',
  },
  space: [0, 4, 8, 16],
};

<ThemeProvider theme={theme}>
  <Flex flexWrap="wrap" my={[2, 3]}>
    <Button variant="default">
      Default
    </Button>
    <Button variant="primary">
      Primary
    </Button>
    <Button variant="danger">
      Danger
    </Button>
    <Button variant="dark">
      Dark
    </Button>
    <Button variant="info">
      Info
    </Button>
    <Button variant="light">
      Light
    </Button>
    <Button variant="link">
      Link
    </Button>
    <Button variant="secondary">
      Secondary
    </Button>
    <Button variant="success">
      Success
    </Button>
    <Button variant="warning">
      Warning
    </Button>
  </Flex>
  <Flex flexWrap="wrap" my={[2, 3]}>
    <Button variant="contained-default">
      Contained-Default
    </Button>
    <Button variant="contained-primary">
      Contained-Primary
    </Button>
    <Button variant="contained-danger">
      Contained-Danger
    </Button>
    <Button variant="contained-dark">
      Contained-Dark
    </Button>
    <Button variant="contained-info">
      Contained-Info
    </Button>
    <Button variant="contained-light">
      Contained-Light
    </Button>
    <Button variant="contained-link">
      Contained-Link
    </Button>
    <Button variant="contained-secondary">
      Contained-Secondary
    </Button>
    <Button variant="contained-success">
      Contained-Success
    </Button>
    <Button variant="contained-warning">
      Contained-Warning
    </Button>
  </Flex>
</ThemeProvider>
```

#### Hide Button

```jsx inside Markdown
import { useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { Text } from 'src/elements/typography';
import {
  Box,
  Flex,
} from 'src/elements/grid';
import { Button } from 'src/elements/Button';

const [hide, setHidden] = useState(false);

<ThemeProvider theme={{}}>
  <Button
    onClick={() => setHidden(!hide)}
  >
    Toggle Visibility
  </Button>
  <Flex flexDirection={['column', 'row']}>
    <Box px={1}>
    <Button isHidden={hide}>
      Hideable
    </Button>
    </Box>
  </Flex>
</ThemeProvider>
```

#### Loading Button

```jsx inside Markdown
<Button
  disabled
  showProgress
>
  Loading...
</Button>
```
