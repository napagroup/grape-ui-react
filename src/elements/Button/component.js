import React from 'react';
import PropTypes from 'prop-types';
import { removeSomeProps } from 'src/utils/componentHelpers';
import { Box, Flex } from 'src/elements/grid';
import {
  getProgress,
  getProgressDefaultProps,
  getProgressPropTypes,
} from 'src/elements/Progress';
import { emailHrefString } from 'src/elements/typography/Link/utils';
import { Hideable } from 'src/elements/utils';
import { boxShadow } from 'styled-system';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  flexboxProps,
  layoutProps,
  positionProps,
  spaceProps,
  typography,
} from 'src/utils/styledHelpers';

const propsToTrim = {
  ...flexboxProps.propTypes,
  ...layoutProps.propTypes,
  ...positionProps.propTypes,
  ...spaceProps.propTypes,
  ...typography.propTypes,
  ...getProgressPropTypes,
  boxSizing: PropTypes.any,
  innerBoxProps: PropTypes.object,
  innerFlexProps: PropTypes.object,
  leadingIcon: PropTypes.any,
  leadingIconProps: PropTypes.object,
  variant: PropTypes.string,
};
const propsToTrimForButton = {
  ...propsToTrim,
  ...boxShadow.propTypes,
  ...getProgressDefaultProps,
  bgActiveColor: PropTypes.string,
  bgHoverColor: PropTypes.string,
  contained: false,
  href: PropTypes.string,
  outline: false,
  raised: false,
  target: PropTypes.string,
  to: PropTypes.string,
};

const propsToTrimForLink = {
  ...propsToTrim,
  ...boxShadow.propTypes,
  bgActiveColor: PropTypes.string,
  bgHoverColor: PropTypes.string,
  contained: false,
  outline: false,
  raised: false,
};

const getChildren = options => {
  const {
    children,
    innerBoxProps,
    innerFlexProps,
    leadingIcon,
    leadingIconProps,
    p,
    pl,
    px,
    showProgress,
  } = options;
  if (leadingIcon || showProgress) {
    return (
      <Flex {...innerFlexProps}>
        <Box pr={pl || px || p} {...leadingIconProps}>
          <Hideable hide={showProgress}>
            {leadingIcon}
          </Hideable>
          {getProgress({ showProgress, ...options })}
        </Box>
        <Box {...innerBoxProps}>{children}</Box>
      </Flex>
    );
  }
  return <Box {...innerBoxProps}>{children}</Box>;
};

export const ButtonComponent = ({
  emailHref,
  href,
  to,
  ...props
}) => {
  let output = null;
  const trimmedPropsLink = removeSomeProps(props, Object.keys(propsToTrimForLink));
  const emailLinkHref = emailHrefString(emailHref);
  if (href) {
    output = (
      <a href={href} {...trimmedPropsLink}>
        {getChildren(props)}
      </a>
    );
  } else if (to) {
    const linkProps = {
      to,
    };
    output = (
      <ReactRouterLink {...trimmedPropsLink} {...linkProps}>
        {getChildren(props)}
      </ReactRouterLink>
    );
  } else if (emailLinkHref) {
    return (
      <a href={emailLinkHref} {...trimmedPropsLink}>
        {getChildren(props)}
      </a>
    );
  } else {
    output = (
      <button type="button" {...removeSomeProps(props, Object.keys(propsToTrimForButton))}>
        {getChildren(props)}
      </button>
    );
  }
  return output;
};
ButtonComponent.propTypes = {
  children: PropTypes.any,
  emailHref: PropTypes.shape({
    bcc: PropTypes.string,
    body: PropTypes.string,
    cc: PropTypes.string,
    subject: PropTypes.string,
    to: PropTypes.string,
  }),
  href: PropTypes.string,
  to: PropTypes.string,
};
ButtonComponent.defaultProps = {
  children: null,
  emailHref: {},
  href: null,
  to: null,
};
