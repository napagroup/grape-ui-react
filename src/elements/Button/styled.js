import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  border,
  boxShadow,
  buttonStyle,
  display,
  flexbox,
  fontWeight,
  layout,
  lineHeight,
  maxWidth,
  position,
  space,
  system,
  verticalAlign,
} from 'styled-system';
import {
  borderRadiusCore,
  colorCore,
  defaultStylesBase,
  ellipsisCore,
  fontFamilyCore,
  fontSizeCore,
  fontStyleCore,
  letterSpacingCore,
  POSITION_DEFAULT_VALUE,
  resolveBoxShadow,
  resolveColor,
  textAlignCore,
  textDecorationCore,
  typography,
} from 'src/utils/styledHelpers';
import { Box } from 'src/elements/grid';
import { withHideable } from 'src/elements/utils/Hideable';
import propTypes from '@styled-system/prop-types';
import { activeColorButton, hoverColorButton } from './utils';
import { ButtonComponent } from './component';

const borderButton = props => border({
  ...props,
  border: props.outline ? props.border : '0',
});

const boxShadowButtonMemoized = (value = '01') => props => boxShadow({
  ...props,
  boxShadow: props.raised ? resolveBoxShadow(value) : '',
});

const scaleButton = props => {
  const {
    pb: pbProps,
    pl: plProps,
    pr: prProps,
    pt: ptProps,
  } = props;
  let pb = pbProps;
  let pl = plProps;
  let pr = prProps;
  let pt = ptProps;
  const { sm, lg } = props;
  if (lg) {
    pb = 2;
    pl = 4;
    pr = 4;
    pt = 2;
  } else if (sm) {
    pb = 0;
    pl = 2;
    pr = 2;
    pt = 0;
  }
  const nextProps = {
    ...props,
    pb,
    pl,
    pr,
    pt,
  };
  return space(nextProps);
};

const Button = styled(withHideable(ButtonComponent))`
  ${system({ boxSizing: true })}
  ${borderButton}
  ${borderRadiusCore}
  ${boxShadowButtonMemoized()}
  ${buttonStyle}
  ${colorCore}
  ${flexbox}
  ${fontFamilyCore}
  ${fontSizeCore}
  ${fontStyleCore}
  ${fontWeight}
  ${layout}
  ${letterSpacingCore}
  ${lineHeight}
  ${position}
  ${scaleButton}
  ${textAlignCore}
  ${textDecorationCore}
  outline: 0;
  text-transform: uppercase;
  &:hover {
    cursor: pointer;
    ${hoverColorButton};
  }
  &:active {
    ${boxShadowButtonMemoized('02')};
    ${activeColorButton};
  }
  &[disabled] {
    opacity: 0.4;
    pointer-events: none;
    cursor: not-allowed;
  }
  > ${Box} {
    ${borderRadiusCore}
    ${display}
    ${ellipsisCore}
    ${maxWidth}
    ${verticalAlign}
  }
`;

Button.propTypes = {
  /** Defines the active background color for the control. */
  bgActiveColor: PropTypes.string,
  /** Defines the hover background color for the control. */
  bgHoverColor: PropTypes.string,
  ...boxShadow.propTypes,
  /** Makes the button a "contained" button.
   * @see See [Material Design/Components/Buttons/Contained Button](https://material.io/components/buttons/#contained-button) for more on this style. */
  contained: PropTypes.bool,
  ...display.propTypes,
  /** Define properties for an email.  Fill in props and the control will generate the proper string. */
  emailHref: PropTypes.shape({
    /** Sets the BCC line. Can be comma-seperatred list. */
    bcc: PropTypes.string,
    /** Sets the Body. */
    body: PropTypes.string,
    /** Sets the CC line. Can be comma-seperatred list. */
    cc: PropTypes.string,
    /** Sets the Subject Line. */
    subject: PropTypes.string,
    /** Sets who to send it to. Can be comma-seperatred list. */
    to: PropTypes.string,
  }),
  ...fontWeight.propTypes,
  /** Will use an anchor tag instead of a button tag. */
  href: PropTypes.string,
  ...lineHeight.propTypes,
  ...maxWidth.propTypes,
  /** Passes props to the `Box` child. */
  innerBoxProps: PropTypes.object,
  /** Passes props to the `Flex` child that appears when a `leadingIcon` is provided. */
  innerFlexProps: PropTypes.object,
  /** Hides component */
  isHidden: PropTypes.bool,
  /** Placeholder spot meant for icons. */
  leadingIcon: PropTypes.any,
  /** Passes props to the leading icon's container. */
  leadingIconProps: PropTypes.object,
  /** Makes the button a "contained" button.
   * @see See [Material Design/Components/Buttons/Outlined Button](https://material.io/components/buttons/#outlined-button) for more on this style. */
  outline: PropTypes.bool,
  ...position.propTypes,
  /** Makes the button a "raised" button.
   * @see See [Material Design/Components/Buttons/Hierarchy and Placement](https://material.io/components/buttons/#hierarchy-placement) for more on this style. */
  raised: PropTypes.bool,
  ...space.propTypes,
  /** Will use react-router's Link component. You will still need to wrap this in a Router component. */
  to: PropTypes.string,
  ...typography.propTypes,
  ...propTypes.border,
  ...propTypes.flexbox,
  ...propTypes.layout,
  ...propTypes.position,
};

Button.defaultProps = {
  ...defaultStylesBase,
  bg: null,
  bgActiveColor: null,
  bgHoverColor: null,
  border: `1px solid ${resolveColor('borderColor')}`,
  borderRadius: '',
  boxSizing: 'border-box',
  color: null,
  contained: false,
  display: 'inline-block',
  emailHref: {},
  href: null,
  innerBoxProps: {},
  innerFlexProps: {},
  isHidden: false,
  leadingIcon: null,
  leadingIconProps: {},
  m: 1,
  maxWidth: '100%',
  outline: false,
  pb: 1,
  pl: 3,
  position: POSITION_DEFAULT_VALUE,
  pr: 3,
  progressProps: {
    progressType: 'circular',
  },
  pt: 1,
  raised: false,
  textAlign: 'center',
  textDecoration: 'none',
  to: null,
  verticalAlign: 'top',
};

/** @component */
export { Button };
