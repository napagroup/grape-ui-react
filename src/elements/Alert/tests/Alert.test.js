/* eslint-disable react/prop-types */

import React from 'react';
import * as ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Box, Flex } from 'src/elements/grid';
import { Button } from 'src/elements/Button';
import { Alert } from '..';

describe('Alert - kitchen sink snapshot', () => {
  let container;
  beforeEach(() => {
    container = document.createElement('div');
  });
  it('should return Alert with all components & styling', () => {
    const UndoButton = () => <Button color="white">Undo</Button>;
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Alert
            alertAction={<UndoButton />}
          >
            <Flex>
              <Box>Uh oh</Box>
              <Box>Spaghettios</Box>
            </Flex>
          </Alert>
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});

describe('Alert - Variant Example', () => {
  let container;
  beforeEach(() => {
    container = document.createElement('div');
  });
  it('should return Alert with variant styling', () => {
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Alert
            variant="danger"
          >
            Variant Alert.
          </Alert>
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
  it('should return Alert with custom variant styling', () => {
    const colorThemeProps = {
      colors: {
        brandDanger: 'red',
      },
    };
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={colorThemeProps}>
          <Alert
            variant="danger"
          >
            Variant Alert with Custom Theme.
          </Alert>
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});
