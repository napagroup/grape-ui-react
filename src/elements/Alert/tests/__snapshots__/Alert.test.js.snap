// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert - Variant Example should return Alert Toast Container 1`] = `
.c0 {
  box-sizing: border-box;
  position: fixed;
}

.c1 {
  cursor: pointer;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  justify-self: center;
  left: 50vw;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c1 .Toastify__progress-bar {
  -webkit-animation: kJyxGF linear 1;
  animation: kJyxGF linear 1;
}

.c1 .Toastify__toast-body > div {
  margin: 0;
}

.c1 .Toastify__toast-body + button {
  position: absolute;
  top: 50%;
  right: 0;
  -webkit-transform: translateY(-50%) translateY(-6px);
  -ms-transform: translateY(-50%) translateY(-6px);
  transform: translateY(-50%) translateY(-6px);
}

<div
  id="custom"
>
  <style>
    
  @keyframes zoomInKeyframes {
    from {
      opacity: 0;
      transform: scale3d(.9, .9, .9);
    }

    to {
      opacity: 1;
    }
  }

  @keyframes zoomOutKeyframes {
    from {
      opacity: 1;
    }

    50% {
      transform: scale3d(.9, .9, .9);
    }

    to {
      opacity: 0;
    }
  }

  .zoomIn { animation-name: zoomInKeyframes; }
  .zoomOut { animation-name: zoomOutKeyframes; }

  </style>
  <div
    class="c0 c1"
  >
    <div
      class="Toastify"
    />
  </div>
</div>
`;

exports[`Alert - Variant Example should return Alert with custom variant styling 1`] = `
.c3 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c0 {
  box-sizing: border-box;
  border-radius: 4px;
  display: inline-block;
  max-width: 1px;
  min-width: 300px;
  overflow: hidden;
  margin: 8px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
}

.c1 {
  background-color: #f99;
  color: red;
}

@media screen and (min-width:40em) {
  .c0 {
    max-width: 600px;
  }
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c0 {
    margin: 16px;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        Variant Alert with Custom Theme.
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert - Variant Example should return Alert with variant styling 1`] = `
.c3 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c0 {
  box-sizing: border-box;
  border-radius: 4px;
  display: inline-block;
  max-width: 1px;
  min-width: 300px;
  overflow: hidden;
  margin: 8px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
}

.c1 {
  background-color: #fcc9c5;
  color: hsl(4.1,89.6%,58.4%);
}

@media screen and (min-width:40em) {
  .c0 {
    max-width: 600px;
  }
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c0 {
    margin: 16px;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <span
        class="c3"
      >
        Variant Alert.
      </span>
    </div>
  </div>
</div>
`;

exports[`Alert - kitchen sink snapshot should return Alert with all components & styling 1`] = `
.c1 {
  box-sizing: border-box;
  border-radius: 4px;
  display: inline-block;
  max-width: 1px;
  min-width: 300px;
  overflow: hidden;
  margin: 8px;
}

.c5 {
  box-sizing: border-box;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  box-sizing: border-box;
  border: 0;
  border-radius: 4px;
  color: hsl(0,0%,98%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c6:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c6:active {
  background-color: hsl(0,0%,100%);
}

.c6[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c6 > .c0 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

.c2 {
  background-color: hsl(0,0%,25%);
  color: #fff;
}

@media screen and (min-width:40em) {
  .c1 {
    max-width: 600px;
  }
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c1 {
    margin: 16px;
  }
}

<div>
  <div
    class="c0 c1 c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c0 c5"
        >
          Uh oh
        </div>
        <div
          class="c0 c5"
        >
          Spaghettios
        </div>
      </div>
      <div
        class="c0 c5"
      >
        <button
          class="c6"
          type="button"
        >
          <div
            class="c0 c5"
          >
            Undo
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
